lib/CrossTU/CrossTranslationUnit.cpp

```diff
@@ -349,6 +349,7 @@ llvm::Expected<ASTUnit *> CrossTranslationUnitContext::loadExternalAST(
           ASTUnit::LoadEverything, Diags, CI.getFileSystemOpts()));
       Unit = LoadedUnit.get();
       FileASTUnitMap[ASTFileName] = std::move(LoadedUnit);
+      SrcFileASTUnitMap[Unit->getOriginalSourceFileName()] = Unit;
```

Comment:
I wonder if this would work well if we have different files (on different paths) with the same name. Shouldn't we make sure this is an absolute path for safety?

Reply:
If there are different files with identical name that does not work. The assumption is that this "original filename" is the same that is contained in the `SourceLocation` for the main (in that `ASTUnit`) AST file. An assert can be added here to check that the map has no entry already.